server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -56,6 +56,7 @@ public class NRTReplicationEngine extends Engine {
     private final CompletionStatsCache completionStatsCache;
     private final LocalCheckpointTracker localCheckpointTracker;
     private final WriteOnlyTranslogManager translogManager;
+    private final boolean shouldCommit;
```

Comment:
nit: `shouldCommit` -> `remoteStoreEnabled`. The later is self explanatory.

Reply:
This was named `shouldCommit` from `remoteStoreEnabled` from a previous conversation with Marc on this PR - to signify that the configuration on the engine is "should I commit" vs "am I remote store"

Reply:
Yeah the idea was to have this passed as config to the engine that shouldn't care about the source of replication.  Given we get the entire settings object it felt cleaner to set this in the ctor.

Reply:
Sounds good!